### DISCUSSION TITLE
API "URL Not found"

### DISCUSSION THREAD

**Discussion by phils01UK** [ZoneMinder/zoneminder#4107] [Q&A]
I am probably missing something very obvious, but on my new ZM install I have enabled API with OPT_USE_API in the options page. But when I access http://ip/zm/api/host/login.json I get a 404/page not found.
I dont remember having to do anythgin else last time I installed a ZM server.
Can anyone point me int he right direction as to why I am getttign 404s
Thanks

**Comment by connortechnology** [comment]
debian and ubuntu official packages  do not correctly setup the api in apache.  zoneminder.conf needs these lines appended 
# For better visibility, the following directives have been migrated from the
# default .htaccess files included with the CakePHP project.
# Parameters not set here are inherited from the parent directive above.
<Directory "/usr/share/zoneminder/www/api">
   RewriteEngine on
   RewriteRule ^$ app/webroot/ [L]
   RewriteRule (.*) app/webroot/$1 [L]
   RewriteBase /zm/api
</Directory>

<Directory "/usr/share/zoneminder/www/api/app">
   RewriteEngine on
   RewriteRule ^$ webroot/ [L]
   RewriteRule (.*) webroot/$1 [L]
   RewriteBase /zm/api
</Directory>

<Directory "/usr/share/zoneminder/www/api/app/webroot">
    RewriteEngine On
    RewriteCond %{REQUEST_FILENAME} !-d
    RewriteCond %{REQUEST_FILENAME} !-f
    RewriteRule ^ index.php [L]
    RewriteBase /zm/api
</Directory>

**Reply by phils01UK** [reply]
Thank you - I will try these.
Interestignly, my previous Debian install has no specific apache conf for ZM, and no rewrite rules for the API.
